runtime/compiler/control/JITClientCompilationThread.cpp

```diff
@@ -44,7 +44,7 @@
 #include "vmaccess.h"
 
 extern TR::Monitor *assumptionTableMutex;
-
+bool serializeClass = false;
```

Comment:
This global can create problems with multithreading.
Please define a local as close as possible to the usage.